flixel/input/actions/FlxActionInputDigital.hx

```diff
@@ -0,0 +1,294 @@
+	{
+		return switch (inputID)
+		{
+			#if !FLX_NO_MOUSE
+			case FlxMouseButtonID.LEFT  : switch (trigger)
+			{
+				case PRESSED:        FlxG.mouse.pressed || FlxG.mouse.justPressed;
+				case RELEASED:      !FlxG.mouse.pressed || FlxG.mouse.justReleased;
+				case JUST_PRESSED:   FlxG.mouse.justPressed;
+				case JUST_RELEASED:  FlxG.mouse.justReleased;
+			}
+			case FlxMouseButtonID.MIDDLE: switch (trigger)
+			{
+				case PRESSED:        FlxG.mouse.pressedMiddle || FlxG.mouse.justPressedMiddle;
+				case RELEASED:      !FlxG.mouse.pressedMiddle || FlxG.mouse.justReleasedMiddle;
+				case JUST_PRESSED:   FlxG.mouse.justPressedMiddle;
+				case JUST_RELEASED:  FlxG.mouse.justReleasedMiddle;
+			}
+			case FlxMouseButtonID.RIGHT : switch (trigger)
+			{
+				case PRESSED:        FlxG.mouse.pressedRight || FlxG.mouse.justPressedRight;
+				case RELEASED:      !FlxG.mouse.pressedRight || FlxG.mouse.justReleasedRight;
+				case JUST_PRESSED:   FlxG.mouse.justPressedRight;
+				case JUST_RELEASED:  FlxG.mouse.justReleasedRight;
+			}
+			#end
+			default: false;
+		}
+	}
+}
+
+class FlxActionInputDigitalSteam extends FlxActionInputDigital
+{
+	/**
+	 * Steam Controller action input for digital (button-like) events
+	 * @param	ActionHandle handle received from FlxSteamController.getDigitalActionHandle()
+	 * @param	Trigger what state triggers this action (PRESSED, JUST_PRESSED, RELEASED, JUST_RELEASED)
+	 * @param	DeviceHandle handle received from FlxSteamController.getConnectedControllers(), or FlxInputDeviceID.ALL / FIRST_ACTIVE
+	 */
+	@:allow(flixel.input.actions.FlxActionSet)
+	private function new(ActionHandle:Int, Trigger:FlxInputState, ?DeviceHandle:Int = FlxInputDeviceID.FIRST_ACTIVE)
+	{
+		super(FlxInputDevice.STEAM_CONTROLLER, ActionHandle, Trigger, DeviceHandle);
+		#if steamwrap
+			steamInput = new FlxInput<Int>(ActionHandle);
+		#else
+			FlxG.log.warn("steamwrap library not installed; steam inputs will be ignored.");
+		#end
+	}
+	
+	override public function check(Action:FlxAction):Bool 
+	{
+		return switch (trigger)
+		{
+			case PRESSED:        steamInput.pressed  || steamInput.justPressed;
+			case RELEASED:      !steamInput.released || steamInput.justReleased;
+			case JUST_PRESSED:   steamInput.justPressed;
+			case JUST_RELEASED:  steamInput.justReleased;
+		}
+	}
+	
+	override public function update():Void 
+	{
+		if (getSteamControllerData(deviceID))
+			steamInput.press();
+		else
+			steamInput.release();
+	}
+	
+	private var steamInput:FlxInput<Int>;
+	
+	private inline function getSteamControllerData(controllerHandle:Int):Bool
+	{
+		if (controllerHandle == FlxInputDeviceID.FIRST_ACTIVE)
+		{
+			controllerHandle = FlxSteamController.getFirstActiveHandle();
+		}
+		
+		var data = FlxSteamController.getDigitalActionData(controllerHandle, inputID);
+		
+		return (data.bActive && data.bState);
+	}
+}
+
+class FlxActionInputDigitalIFlxInput extends FlxActionInputDigital
```

Comment:
Heh. What's funny is that this is a direct result of the naming convention I've established and it just turns out to be a really ridiculous edge case.

"a digital input for a flxAction, where the input is of type IFlxInput"

I'm not entirely sure whether I should rework the entire naming convention because "FlxActionInputDigital" gets a bit unwieldy, or what. 

Is it our policy to stick "Flx" on literally everything, or drop it for certain situations? And are we at all consistent in that? It'd be painful to remove it for existing classes, but what about new ones? `ActionInputDigital<Something>` would certainly be a bit less unwieldy.

However, it doesn't look easy to drop "Action" because without it "Input" becomes ambiguous because there's so many other contexts in which "Input" is used.

Lastly, I do think it's important to distinguish between Digital and Analog action inputs, but maybe that should come at the beginning rather than squashed towards the end?

`DigitalActionInput<Something>`

Anyways, open to ideas.